File: flutter17_mobile/lib/models/shopping_cart.dart
import 'package:flutter17_mobile/models/cart_item.dart';
import 'package:json_annotation/json_annotation.dart';

part 'shopping_cart.g.dart';

@JsonSerializable()
class ShoppingCart {
  int? id;
  int? customerId;
  DateTime? dateCreated;
  List<CartItem>? cartItems;
  double? totalAmount;

  ShoppingCart({
    this.id,
    this.customerId, 
    this.dateCreated,
    this.cartItems,
    this.totalAmount,
  });

  factory ShoppingCart.fromJson(Map<String, dynamic> json) => _$ShoppingCartFromJson(json);

  Map<String, dynamic> toJson() => _$ShoppingCartToJson(this);
}

File: angular17_web/src/app/models/shopping-cart/shopping.cart.ts
import { CartItem } from '../cart-item/cart.item';

export class ShoppingCart {
  id!: number;
  customerId!: number;
  dateCreated!: Date;
  totalAmount?: number;
  cartItems: CartItem[] = [];

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.customerId = data.customerId;
    this.dateCreated = data.dateCreated ? new Date(data.dateCreated) : undefined as any;
    this.totalAmount = data.totalAmount;
    this.cartItems = Array.isArray(data.cartItems)
      ? data.cartItems.map((x: any) => new CartItem(x))
      : [];
  }
}
